DBM-1743: Provide several agent configuration examples (#15900)
DataDog/documentation: layouts/shortcodes/dbm-postgres-agent-config-examples.md

```yaml
init_config:
instances:
  - dbm: true
    host: localhost
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    custom_queries:
    - metric_prefix: employee
      query: SELECT age, salary, hours_worked, name FROM hr.employees;
      columns:
        - name: age
          type: gauge
        - name: salary
           type: gauge
        - name: hours
           type: count
        - name: name
           type: tag
      tags:
        - 'table:employees'
```

### Monitoring multiple databases 
To specify the name of the database to connect to, use `dbname`. If omitted, the default system `postgres` database is queried. Additionally, configure `dbstrict` to `true` if you only want to gather metrics from the database provided in the `dbname` option.

```yaml
init_config:
instances:
  - dbm: true
    host: localhost
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    dbname: foo  # Default: `postgres`
    # dbstrict: true 
  - dbm: true

>     host: localhost

Reviewer: If they are monitoring mult dbs on the same host we don't want them to put `dbm:true` twice, since dbm pulls metrics from a global view 

I wrote some stuff about this in our internal support FAQ, that you can use for this https://datadoghq.atlassian.net/wiki/spaces/DM/pages/2645229741/PostgreSQL+FAQ#How-to-monitor-multiple-databases-with-dbm-&-the-default-pg-integration

Author: ah good to know, thanks!

### One agent connecting to multiple hosts
It is common to configure a single Agent host to connect to multiple remote database instances (see [Agent installation architectures](/database_monitoring/architecture/) for DBM). To connect to multiple hosts, create an entry for each host in the Postgres integration config.

> In these cases, it is recommended to limit to the number of instances per Agent to a maximum of 10 database instances to guarantee reliable performance.

Reviewer 2: ```suggestion
In these cases, Datadog recommends limiting the number of instances per Agent to a maximum of 10 database instances to guarantee reliable performance.
```

```yaml
init_config:
instances:
  # This instance is the "DBM" instance. It will connect to the
  # `postgres` database and send DBM telemetry from all databases
  - dbm: true
    host: products-primary.123456789012.us-east-1.rds.amazonaws.com
    port: 5432
    username: datadog
    password: '<PASSWORD>'
  # This instance only collects data from the `on_sale` database
  # and collects relation metrics from tables prefixed by "2022_"
  - host: products-primary.123456789012.us-east-1.rds.amazonaws.com
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    dbname: on_sale
    dbstrict: true
    relations:
      - relation_regex: 2022_.*
        relkind:
          - r
          - i
  # This instance only collects data from the `inventory` database
  # and collects relation metrics only from the specified tables
  - host: products-primary.123456789012.us-east-1.rds.amazonaws.com
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    dbname: inventory
    dbstrict: true
    relations:
      - relation_name: products
      - relation_name: external_seller_products
```

### Working with hosts through a proxy

> If the Agent must connect through a proxy such as the [Cloud SQL Auth proxy](https://cloud.google.com/sql/docs/mysql/connect-admin-proxy), all telemetry is tagged with the hostname of the proxy rather than the database instance. Utilize the `reported_hostname` option to set a custom override of the hostname detected by the Agent.

Reviewer 2: ```suggestion
If the Agent must connect through a proxy such as the [Cloud SQL Auth proxy](https://cloud.google.com/sql/docs/mysql/connect-admin-proxy), all telemetry is tagged with the hostname of the proxy rather than the database instance. Use the `reported_hostname` option to set a custom override of the hostname detected by the Agent.
```

### One agent connecting to multiple hosts

> To connect to multiple hosts, create an entry for each host in the Postgres integration config.

Reviewer 3: Same comments for MySQL above.

```yaml
init_config:
instances:
  - dbm: true
    host: localhost_A
    port: 5432
    username: datadog
    password: '<PASSWORD>'
  - dbm: true
    host: localhost_B
    port: 5432
    username: datadog
    password: '<PASSWORD>'
  - dbm: true
    host: localhost_C
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    [...]
```

### Running custom queries

> To collect custom metrics, use the `custom_queries` option. See the sample [postgres.d/conf.yaml](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/data/conf.yaml.example) for more details.

Reviewer 3: Same comments from MySQL.

```yaml
init_config:
instances:
  - dbm: true
    host: localhost
    port: 5432
    username: datadog
    password: '<PASSWORD>'
    custom_queries:
    - metric_prefix: ddprefix
      query: SELECT age, salary, name FROM foo;
      columns:
        - name: foo_age
          type: gauge
        - name: foo_salary
          type: count
        - name: foo_name
          type: tag
      tags:
        - query:custom
```

### Monitoring multiple databases 

> To specify the name of the database to connect to, use `dbname`. If omitted, the default system `postgres` database is queried. Additionally, configure `dbstrict` to `true` if you only want to gather metrics from the database provided in the `dbname` option.

Reviewer 3: This section needs to be more opinionated. This is the most confusing aspect of our configuration. It should clearly explain to the customer how they should configure it, what they get by doing so, and why. 

```Additionally, configure `dbstrict` to `true` if you only want to gather metrics``` puts too much burden on the customer to understand the highly confusing options. These examples should be copy-paste-replace and work without any thought on the customer's part.

        - 'table:employees'
```

### Monitoring multiple databases with DBM and the default integration

> To specify the name of the database to connect to, use `dbname`. If omitted, the default system `postgres` database is queried.

Reviewer: might want to add some words here to explain why only one of the instances has `dbm: true` I have some words about this here https://datadoghq.atlassian.net/wiki/spaces/DM/pages/2645229741/PostgreSQL+FAQ#How-to-monitor-multiple-databases-with-dbm-&-the-default-pg-integration

this part is important for customers to understand 

> If you wish to collect the [generic postgres integration]<INSERT LINK HERE> metrics from logical databases, other than the default `postgres` db, you must list them each in their instance config. **Note:** you should list each logical db without setting `dbm: true`, and then add one final instance with `dbm:true` so that we don't "double monitor" the databases. This is because database-montoring collects metrics from global views like `pg_stat_statements` and `pg_stat_activity`.
